addon/components/rdf-input-fields/files.js

```diff
@@ -166,4 +177,21 @@ export default class RdfInputFieldsFilesComponent extends InputFieldComponent {
     this.files.removeObject(fileField);
     // general validation of the field is handled by onStoreUpdate()
   }
+
+  downloadAsZip = dropTask(async () => {
+    const promises = this.files.map((file) => {
+      return fetch(file.record.downloadLink);
```

Comment:
One thing I forgot, and just noticed.. `fetch` doesn't consider 40* responsen to be errors, so 404s don't make the promise reject either 😄.

So the zip will contain a file with the correct name, but its contents will be the '404, not found' text.